.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+name: run script.py
+
+on:
+  schedule:
+    - cron: '0 */2 * * *' # Every Two Hours 
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: checkout repo content
+        uses: actions/checkout@v2 # checkout the repository content to github runner
+
+      - name: setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9' # install the python version needed
+          
+      - name: install python packages
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          
+      - name: execute py script # run main.py
```

Comment:
You're not executing main.py but a file called script py. Either rename the file or rename the comment (better to rename the file since main.py is standard)